### DISCUSSION TITLE
Does nearest compute XYZ distance only or also feature distance ?

### DISCUSSION THREAD

**Discussion by QuanticDisaster** [pyg-team/pytorch_geometric#2856] [Q&A] [answered]
I think the title is self explanatory.

When using torch_geometric.nn.nearest with x and y having additionnal features (such as intensity) does the "distance" computed is only with XYZ coordinates or also with features ?

**Comment by QuanticDisaster** [comment, marked as answer]
Alright I tested it myself and the result is that the distance is computed in the feature dimension. Maybe it should be specified in the doc just to be sure !

**Reply by rusty1s** [reply]
The distance is indeed computed based on all features in `x` and `y`.
